src/libcore/ops.rs

```diff
@@ -64,6 +64,50 @@
 //!
 //! See the documentation for each trait for a minimum implementation that
 //! prints something to the screen.
+//!
+//! The `Fn`, `FnMut`, and `FnOnce` traits are implemented by types that can be
+//! invoked like functions. Note that `Fn` takes `&self`, `FnMut` takes `&mut
+//! self` and `FnOnce` takes `self`. These correspond to the three kinds of
+//! methods that can be invoked on an instance: call-by-reference,
+//! call-by-mutable-reference, and call-by-value. The most common use of these
+//! traits is to act as bounds to higher-level functions that take functions or
+//! closures as arguments.
+//!
+//! Taking a `Fn` as a parameter:
+//!
+//! ```rust
+//! fn call_with_one<F>(func: F) -> usize
+//!     where F: Fn(usize) -> usize {
+//!     func(1)
+//! }
+//!
+//! let double = &|x| x + x;
+//! assert_eq!(call_with_one(double), 2);
+//! ```
+//!
+//! Taking a `FnMut` as a parameter:
+//!
+//! ```rust
+//! fn call_with_one<F>(mut func: F) -> usize
+//!     where F: Fn(usize) -> usize {
+//!     func(1)
+//! }
+//!
+//! let double = &mut |x| x + x;
```

Comment:
I think this example could be better; this isn't actually a `FnMut`, I don't think, but a `&mut Fn`. Which then coerces.

A `FnMut` would be something that captures a mutable value and mutates it